src/app/hooks/useViewTracker/index.jsx

```diff
@@ -18,10 +18,14 @@ const useViewTracker = (props = {}) => {
   const url = path(['url'], props);
   const timer = useRef(null);
   const [viewSent, setViewSent] = useState(false);
-  const { enabled: eventTrackingIsEnabled } = useToggle('eventTracking');
+  const { enabled: eventTrackingIsEnabled, value } = useToggle('eventTracking');
+  const isExcluded =
+    typeof value === 'string'
+      ? value?.trim().split(',').includes(componentName)
+      : false;
```

Comment:
Interesting, exactly the same algorithm in both event and view tracking, I'd seriously consider extracting this logic out into a shared `resolveTrackingConfig` or something like that, it could have an API like this:
`const { enabled, excludedComponents } = resolveTrackingConfig('eventTracking');`

Although the logic is only in two places you have a lot of duplicated tests between the view and click tracker, you could avoid this by testing all the code around the value being: null, empty, wrong data type etc for the `resolveTrackingConfig` component and then assume it will always return a valid array in the click and view tracking tests. Give me a shout if you'd like me to explain this more.

Reply:
Great idea, I was thinking of making a utilities directory today to help with testing the tracking on each component, but I hadn't thought of this. Thanks @andrewscfc!

Reply:
Created a new hook, `useTrackingToggle`, which takes the component name and returns whether or not it's enabled in the current environment. Let me know what you think, cheers!